hail/src/main/scala/is/hail/expr/ir/Parser.scala

```diff
@@ -936,13 +937,13 @@ object IRParser {
         val rowExpr = ir_value_expr(newEnv)(it)
         MatrixAggregateRowsByKey(child, entryExpr, rowExpr)
       case "MatrixRead" =>
-        val typ = opt(it, matrix_type_expr)
+        val requestedType = opt(it, matrix_type_expr)
         val dropCols = boolean_literal(it)
         val dropRows = boolean_literal(it)
         val readerStr = string_literal(it)
-        implicit val formats = MatrixReader.formats
+        implicit val formats = MatrixReader.formats + new MatrixBGENReaderSerializer(env)
```

Comment:
can we add this to MatrixReader.formats?

Reply:
I don't think so. I tried to parameterize the formats by the env, but it's also used by Pretty where there is no environment.

Reply:
oh, I see. I missed the part where the BGEN reader depends on the env.